src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNSettings.java

```diff
@@ -367,4 +387,48 @@ public void setClusterService(ClusterService clusterService) {
         this.clusterService = clusterService;
     }
 
+    static class SpaceTypeValidator implements Setting.Validator<String> {
+
+        private Set<String> types = SpaceTypes.getValues();
+
+        @Override public void validate(String value) {
+            if (!types.contains(value.toLowerCase())){
```

Comment:
Why is this case insensitive?

Reply:
I intended to set it case insensitive, e.g. L2, Cosinesimil is also valid string. But I can remove this if not necessary.

Reply:
I think case insensitive is good